parser-typechecker/src/Unison/TermParser.hs

```diff
@@ -760,50 +791,10 @@ importp = do
       suffix <- L.payload <$> suffixes
       pure (suffix, Name.joinDot (L.payload prefix) suffix)
 
---module Monoid where
---  -- we replace all the binding names with Monoid.op, and
---  -- if `op` is free in the body of any binding, we replace it with `Monoid.op`
---  op : Monoid a -> (a -> a -> a)
---  op m = case m of Monoid
-
 data BlockElement v
   = Binding ((Ann, v), Term v Ann)
+  | DestructuringBind (Ann, Term v Ann -> Term v Ann)
   | Action (Term v Ann)
-  | Namespace String [BlockElement v]
-
-namespaceBlock :: Var v => P v (BlockElement v)
-namespaceBlock = do
-  _ <- reserved "namespace"
-  -- need a version of verifyRelativeName that takes a `Token Name`
-  name <- verifyRelativeName importWordyId
-  let statement = (Binding <$> binding) <|> namespaceBlock
-  _ <- openBlockWith "where"
-  elems <- sepBy semi statement
-  _ <- closeBlock
-  pure $ Namespace (Name.toString $ L.payload name) elems
-
-toBindings :: forall v . Var v => [BlockElement v] -> [((Ann,v), Term v Ann)]
-toBindings b = let
-  expand (Binding ((a, v), e)) = [((a, Just v), e)]
-  expand (Action e) = [((ann e, Nothing), e)]
-  expand (Namespace name bs) = scope name $ expand =<< bs
-  v `orBlank` i = fromMaybe (Var.nameds $ "_" ++ show i) v
-  finishBindings bs =
-    [((a, v `orBlank` i), e) | (((a,v), e), i) <- bs `zip` [(1::Int)..]]
-
-  scope :: String -> [((Ann, Maybe v), Term v Ann)]
-                  -> [((Ann, Maybe v), Term v Ann)]
-  scope name bs = let
-    vs :: [Maybe v]
-    vs = snd . fst <$> bs
-    prefix :: v -> v
-    prefix v = Var.named (Text.pack name `mappend` "." `mappend` Var.name v)
-    vs' :: [Maybe v]
-    vs' = fmap prefix <$> vs
-    substs = [ (v, Term.var () v') | (Just v, Just v') <- vs `zip` vs' ]
-    sub = ABT.substsInheritAnnotation substs
-    in [ ((a, v'), sub e) | (((a,_),e), v') <- bs `zip` vs' ]
-  in finishBindings (expand =<< b)
```

Comment:
got to delete some code that was only relevant to hacky namespace implementation